File: solutions/problem_0326_power_of_three/solution.py
"""
Problem 326: Power of Three
==========================
Difficulty: Easy
Tags: Math, Recursion
Pattern: Math

Time Complexity:  O(n) or O(sqrt(n))
Space Complexity: O(1)
"""

class Solution:
    def isPowerOfThree(self, n: int) -> bool:
        # Mathematical approach
        result = 0
        x = n
        while x != 0:
            result = result * 10 + x % 10
            x //= 10 if isinstance(x, int) else 1
        return result

File: solutions/problem_0326_power_of_three/solution.cpp
/*
 * Problem 326: Power of Three
 * ==========================
 * Difficulty: Easy
 * Tags: Math, Recursion
 * Pattern: Math
 *
 * Time Complexity:  O(n) or O(sqrt(n))
 * Space Complexity: O(1)
 */

#include <string>
#include <vector>
using namespace std;

class Solution {
public:
    bool isPowerOfThree(int n) {
        // Mathematical approach
        long long result = 0;
        int x = n;
        while (x != 0) {
            result = result * 10 + x % 10;
            x /= 10;
        }
        return (int)result;
    }
};
